modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/NotificationsSettingsFragment.kt

```diff
@@ -445,6 +402,48 @@ class NotificationsSettingsFragment :
         }
     }
 
+    private fun setupEnabledNotifications() {
+        launch(Dispatchers.Default) {
+            val enabled = settings.notifyRefreshPodcast.flow.value
+
+            launch(Dispatchers.Main) {
+                enabledPreference?.isChecked = enabled
+                enabledPreferences(enabled)
+
+                enabledPreference?.setOnPreferenceChangeListener { _, newValue ->
+                    val checked = newValue as Boolean
+                    settings.notifyRefreshPodcast.set(checked)
```

Comment:
Apart from moving this entire method to be run in `onResume` instead of only when the fragment was created, the only change to the logic was to add this `settings.notifyRefreshPodcast.set(checked)` line so that we now persist the enabled setting.